test/test_torch.py

```diff
@@ -12151,6 +12151,17 @@ def test_ceil_out_mismatch(self, device):
         b = torch.randn(1, device=device)
         self.assertRaises(RuntimeError, lambda: torch.ceil(a, out=b))
 
+    # Test for (some of the) unary funcs' implicit dtype promotion
+    # Check https://github.com/pytorch/pytorch/pull/33322 for more details
+    @dtypes(torch.int8, torch.int16, torch.int32, torch.int64, torch.bool)
+    def test_intbool_to_float_upcasting(self, device, dtype):
+        op_list = ["acos", "asin", "ceil", "expm1", "floor", "log", "log10", "log1p", "log2", \
+                   "sin", "cos", "sinh", "sqrt", "trunc"]
+        for op_name in op_list:
+            x = torch.tensor([2], dtype=dtype, device=device)
+            out = getattr(torch, op_name)(x)
+            # Assert all int and bool dtypes are promoted to default floating dtype
+            self.assertIs(out.dtype, torch.get_default_dtype())
```

Comment:
This adds generic test for the floating unary ops. There are a few ops left to integrate with this type promotion PR, and the list in this test code will also be updated in the next commit.

Reply:
seems good to check that the default dtype is actually respected, i.e. test both float and double default dtypes.

Reply:
Thank you for the suggestion @gchanan , I'll add that in the next commit(s).

Reply:
This should be solved here (https://github.com/pytorch/pytorch/pull/33322/commits/df7f5c60b3d9056c9d35c9e50fa5f56fc011b00d). Please let me know if it doesn't, or if you have any more suggestions. Thanks! :)

Reply:
This has been made simpler by using `@float_double_default_dtype`.